### PR TITLE
Change kytos daemon

### PR DESCRIPTION
**Pull request by macartur**

### CODE REVIEW

File: kytos/core/daemon.py [hunk 1]
```diff
@@ -1,104 +1,32 @@
...
+
+    def run(self):
+        """Run the controller and keep alive."""
+        try:
+            self.controller.start()
+        except SystemExit as exception:
+            self.controller.log.error(exception)
+            self.controller.log.info("Kytos daemon start aborted.")
+        while True:
+            pass
```
Comment (diraol) [inline kytos/core/daemon.py]: Is this `while True` right? When is it going to be 'stopped'?
@cemsbr it looks strange to me, do you agree?

File: kytos/core/controller.py [hunk 2]
```diff
@@ -123,7 +123,7 @@ def enable_logs(self):
     def start(self, restart=False):
         """Create pidfile and call start_controller method."""
         self.enable_logs()
-        if not restart:
+        if not restart and self.options.daemon is True:
```
Comment (diraol) [inline kytos/core/controller.py]: `self.options.daemon is True` or `self.options.daemon is False`?

Isn't the `pidfile` automatically created by `KytosDaemon` and we need to manually create it when we are not on the `daemon` mode?

**Review by diraol — COMMENTED** [review]
@macartur is there any relevant information that worth to be added to the CHANGELOG?

File: kytos/core/daemon.py (hunk 1 above)
Comment (macartur) [reply kytos/core/daemon.py]: This is only to keep the process Alive. When the signal SIGTERM or SIGINT is received the process will call the controller.stop to stop the controller instance and after that, the process will be killed.

Comment (diraol) [reply kytos/core/daemon.py]: Ok! Thanks for the explanation!

**Review by diraol — DISMISSED** [review]

File: kytos/core/daemon.py (hunk 1 above)
Comment (cemsbr) [reply kytos/core/daemon.py]: @macartur, this busy waiting is using 100% of CPU.

**Review by cemsbr — CHANGES_REQUESTED** [review]

**Comment by beraldoleal** [comment]
Can you please, create a better description of this PR ? Or link to any issue ?

File: kytos/core/daemon.py [hunk 3]
```diff
@@ -1,104 +1,35 @@
...
+        while True:
+            sleep(0.02)
+
+    def shutdown(self, signum):
+        """Stop the controller and shutdown the process."""
+        self.controller.stop()
+        while True:
+            if self.controller.status() == 'Stopped':
+                super().shutdown(signum)
+            sleep(0.02)
```
Comment (cemsbr) [inline kytos/core/daemon.py]: @macartur, if sleep is actually required, 0.5 is enough for the user to wait.

**Review by cemsbr — CHANGES_REQUESTED** [review]

**Review by cemsbr — APPROVED** [review]

**Comment by hdiogenes** [comment]
Needs a description and/or linked issue. Change what and why?

Looks like it conflicts with some changes in [PR 605](https://github.com/kytos/kytos/pull/605).

**Comment by beraldoleal** [comment]
@macartur please provide a better description.

**Comment by macartur** [comment]
The [PR 605](https://github.com/kytos/kytos/pull/605) has the same commits of this, so I will close this PR.

**Pull request closed without merging**